### ISSUE TITLE
[Bug] Ctrl + C not working in color dialog text box

### ISSUE THREAD

**Issue by ivictbor** [devforth/painterro#66]

**Comment by theotimepoisseau-bib** [comment]
Hi,
I'm going to enrich this issue.

I use painterro in a form, not in fullscreen.
So my users had to be able to write in textarea, inputs. etc.

When Painterro is open, it catch keydown events and stop propagation if a key is a keyboard shortcuts. So text writted is incomplete :/

I consider Painterro should ignore events (or not listen) if not having focus, for now I hotpatch the keydown event in my code for ignoring when `event.target !== document.body` because when in an input field the target is the input, not the body.

It's also fix this <kbd>Ctrl</kdb> + <kbd>C</kdb> in color dialog :-)

Here is my hotpatch :

```ts
/**
 * Should be called after instantiation, but before show (show is the method who attach events)
 * 
 * @param ptInstance
 */
function hotPatchPainterro(ptInstance) {
  const bakKeyDown = ptInstance.documentHandlers.keydown;

  function painterroOnKeydown(event) {
    if (event.target !== document.body) return;

    bakKeyDown.call(ptInstance, event);
  }

  ptInstance.documentHandlers.keydown = painterroOnKeydown;
}
```

And this is what it looks like in my React Wrapper Component

```tsx
import * as React from "react";
import Painterro from 'painterro';
import useId from "../../hooks/useId";
import {useEffect, useMemo, useRef} from "react";
import * as _ from "lodash";
import {DrawingZoneProps} from "./IDrawingZone";
import styles from './DrawingZone.module.scss';

// https://github.com/devforth/painterro#configuration
const defaults = {
  activeColor: 'black',
  defaultTool: 'brush',
  hiddenTools: ['close'],
};

export default function DrawingZone(props: DrawingZoneProps) {
  const id = useId('drawing-zone');
  const ref = useRef<any>();

  const onChange = useMemo(() =>_.debounce(props.onChange, 1000), [props.onChange]);

  useEffect(() => {
    ref.current = Painterro({...defaults, id, onChange});
    hotPatchPainterro(ref.current);
    ref.current.show();

    return () => {
      ref.current.hide();
      (ref.current.baseEl as Element).removeAttribute('hidden');
      ref.current.baseEl.innerHTML = '';
      ref.current = null;
    };
  }, [onChange]);

  useEffect(() => {
    ref.current?.show(props.image);
  }, [props.image]);

  return (
    <div id={id} className={styles.container} />
  );
}
```

Notice this lib doesn't give us a destroy method who clean up dom and events, so It's not handy for unmount component.

Best regards

**Comment by ivictbor** [comment]
@theotimepoisseau-bib thanks for pointing to this issue. Yes we should detect an event, maybe check for body event source is good solution
```
event.target !== document.body
```

**Comment by lukyer** [comment]
Hi,

same problem here, when Painterro is shown (not fullscreen, just my div), whole page has broken events. I can't even select and copy text. Workaround is making it a little bit better (I can copy text from inputs) but isn't enough for me (can't select regular text, paste event to page itself without being captured by Painterro etc.)

Thanks for help and fixing this :)

**Comment by ivictbor** [comment]
@lukyer , @theotimepoisseau-bib thank you for following up, fixed in 1.0.34
